src/devices/Ht16k33/SevenSegment.cs

```diff
@@ -0,0 +1,292 @@
+            WriteHex(value[0], value[1], value[2], value[3]);
+        }
+
+        /// <summary>
+        /// Write 4 hexadecimal digits display (from left to right) from a single uint (4 byte) value
+        /// </summary>
+        /// <param name="value">uint value</param>
+        public void WriteHex(uint value) =>
+            WriteHex((byte)(value >> 24), (byte)((value >> 16) & byte.MaxValue), (byte)((value >> 8) & byte.MaxValue), (byte)(value & byte.MaxValue));
+
+        /// <summary>
+        /// Write 4 hexadecimal digits display (from left to right) from a single ushort (2 byte) value
+        /// </summary>
+        /// <param name="value">ushort value</param>
+        public void WriteHex(ushort value) =>
+            WriteHex((byte)(value >> 12), (byte)((value >> 8) & Font.HexadecimalMask), (byte)((value >> 4) & Font.HexadecimalMask), (byte)(value & Font.HexadecimalMask));
+        #endregion
+
+        /// <summary>
+        /// Write 4 characters (from left to right)
+        /// </summary>
+        /// <param name="first">first character</param>
+        /// <param name="second">second character</param>
+        /// <param name="third">third character</param>
+        /// <param name="fourth">fourth character</param>
+        public void WriteChars(char first, char second, char third, char fourth)
+            => Write(Font.GetChar(first), Font.GetChar(second), Font.GetChar(third), Font.GetChar(fourth));
+
+        /// <summary>
+        /// Write integer value as decimal digits (from left to right)
+        /// </summary>
+        /// <param name="value">integer value</param>
+        /// <param name="alignment">alignment on display (left or right, right is default)</param>
+        /// <exception cref="ArgumentOutOfRangeException"><paramref name="value"/> must be between -999..9999</exception>
+        public void WriteInt(int value, Alignment alignment = Alignment.Right)
+        {
+            if (value > 9999 || value < -999)
+            {
+                throw new ArgumentOutOfRangeException(nameof(value), $"{nameof(value)} must be between -999..9999");
+            }
+
+            WriteString(value.ToString(), alignment);
+        }
+
+        /// <summary>
+        /// Write string value to display
+        /// </summary>
+        /// <param name="value">value to display, max 4 characters, or 5 characters if the 3rd character is ':' (this also turns on the center colon), or 6 characters if 1st character is also ':'</param>
+        /// <remarks>Unsupported characters will be replaced by whitespace</remarks>
+        /// <param name="alignment">alignment on display (left or right, right is default)</param>
+        /// <exception cref="ArgumentException"><paramref name="value"/>[2] must be a ':'</exception>
+        /// <exception cref="ArgumentOutOfRangeException"><paramref name="value"/> can contain maximum 5 characters</exception>
+        public void WriteString(string value, Alignment alignment = Alignment.Left)
```

Comment:
I'd prefer if this was a separate utility ideally working with all segment displays